File: androidApp/src/main/kotlin/br/com/qor/androidApp/ui/components/EventDateBadge.kt
package br.com.qor.androidApp.ui.components

private const val TimeLabelLength = 5

private val PtBrMonthAbbreviations = listOf(
    "JAN", "FEV", "MAR", "ABR", "MAI", "JUN", "JUL", "AGO", "SET", "OUT", "NOV", "DEZ",
)

/** design-system.md §4.1's floating date badge — `{month}` (pt-BR, uppercase) / `{day}`. */
data class DateBadgeLabel(val month: String, val day: String)

/**
 * `Event.startsAt` is an ISO-8601 string (`api.md` T25's contract) — parsed by field position
 * rather than a date library, since only the calendar month/day/time are needed here, not
 * timezone-aware arithmetic.
 */
fun formatDateBadge(isoStartsAt: String): DateBadgeLabel {
    val datePart = isoStartsAt.substringBefore('T')
    val (_, monthStr, dayStr) = datePart.split("-")
    return DateBadgeLabel(month = PtBrMonthAbbreviations[monthStr.toInt() - 1], day = dayStr)
}

/** The `{time}` portion of design-system.md §4.1's venue/time row — `HH:mm`, 24h, no timezone conversion. */
fun formatEventTime(isoStartsAt: String): String {
    val timePart = isoStartsAt.substringAfter('T').substringBefore('Z')
    return timePart.take(TimeLabelLength)
}

File: iosApp/iosApp/UI/Components/EventDateBadge.swift
import Foundation

/// design-system.md §4.1's floating date badge — `{month}` (pt-BR, uppercase) / `{day}`.
/// Mirrors Android's `EventDateBadge.kt` (A2) — same pt-BR month-abbreviation table, same
/// field-position ISO-8601 parsing (only calendar month/day/time are needed, not
/// timezone-aware arithmetic).
struct DateBadgeLabel: Equatable {
    let month: String
    let day: String
}

private let ptBrMonthAbbreviations = [
    "JAN", "FEV", "MAR", "ABR", "MAI", "JUN", "JUL", "AGO", "SET", "OUT", "NOV", "DEZ"
]

private let timeLabelLength = 5

/// `Event.startsAt` is an ISO-8601 string (`api.md` T25's contract) — parsed by field position
/// rather than a date library, since only the calendar month/day/time are needed here.
func formatDateBadge(isoStartsAt: String) -> DateBadgeLabel {
    let datePart = isoStartsAt.split(separator: "T", maxSplits: 1).first.map(String.init) ?? isoStartsAt
    let components = datePart.split(separator: "-")
    guard components.count == 3, let monthIndex = Int(components[1]), monthIndex >= 1, monthIndex <= 12 else {
        return DateBadgeLabel(month: "", day: "")
    }
    return DateBadgeLabel(month: ptBrMonthAbbreviations[monthIndex - 1], day: String(components[2]))
}

/// The `{time}` portion of design-system.md §4.1's venue/time row — `HH:mm`, 24h, no timezone conversion.
func formatEventTime(isoStartsAt: String) -> String {
    guard let afterT = isoStartsAt.split(separator: "T", maxSplits: 1).last else { return "" }
    let beforeZ = afterT.split(separator: "Z", maxSplits: 1).first.map(String.init) ?? String(afterT)
    return String(beforeZ.prefix(timeLabelLength))
}
